subprojects/core/src/main/java/org/gradle/api/internal/project/CrossProjectConfigurator.java

```diff
@@ -29,4 +30,6 @@ public interface CrossProjectConfigurator {
 
     void rootProject(ProjectInternal project, Action<? super Project> buildOperationExecutor);
 
+    MutationGuard getMutationGuard();
```

Comment:
Would have added javdoc here but I am still not completely sure the purpose of this mutation guard. 

Further investigation should eventually clear that up, but for now we will leave this un-javadoc'd